scenarios/aws/ec2/vm_run.go

```diff
@@ -44,7 +45,12 @@ func VMRun(ctx *pulumi.Context) error {
 			agentOptions = append(agentOptions, agentparams.WithFakeintake(fakeintake))
 		}
 
-		_, err = agent.NewHostAgent(env.CommonEnvironment, vm, agentOptions...)
+		_, err = agent.NewHostAgent(env.CommonEnvironment, vm)
+		return err
+	}
+
+	if env.UpdaterDeploy() {
```

Comment:
`AgentDeploy` and `UpdaterDeploy` are incompatible, you should check for it and fail.